tests/api/test_bulk_delete.py

```diff
@@ -286,6 +298,48 @@ def test_delete_draft_orders(
         id__in=[order.id for order in orders]).count() == len(orders)
 
 
+def test_fail_to_delete_non_draft_order_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
+    # Set anything but OrderStatus.DRAFT
+    order.status = OrderStatus.CANCELED
+    order.save()
+
+    variables = {'ids': [
+        graphene.Node.to_global_id('OrderLine', order_line.id)
+        for order_line in order_lines]}
+    response = staff_api_client.post_graphql(
+        MUTATION_DELETE_ORDER_LINES, variables,
+        permissions=[permission_manage_orders])
+
+    content = get_graphql_content(response)
+    assert 'errors' in content['data']['draftOrderLinesBulkDelete']
+    assert content['data']['draftOrderLinesBulkDelete']['count'] == 0
+
+
+def test_delete_draft_order_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
```

Comment:
`order.lines`?

Reply:
I was recently caught on this too, but `Order` has overridden method:
```
    def __iter__(self):
        return iter(self.lines.all())
```